ckanext/validation/utils.py

```diff
@@ -138,6 +138,31 @@ def run_sync_validation(resource_data):
         resource_data['_success_validation'] = True
 
 
+def _get_uploaded_resource_path(resource_data):
+    """Get a path for uploaded resource. Supports a default ResourceUpload and
+    ckanext-s3filestore S3ResourceUploader."""
+    upload = uploader.get_resource_uploader(resource_data)
+    path = None
+
+    if isinstance(upload, uploader.ResourceUpload):
+        path = upload.get_path(resource_data['id'])
+    else:
+        try:
+            from ckanext.s3filestore.uploader import S3ResourceUploader
+        except Exception:
+            return path
+
+        if isinstance(upload, S3ResourceUploader):
+            filename = os.path.basename(resource_data["url"])
+            key_path = upload.get_path(resource_data["id"], filename)
```

Comment:
Don't we have a 'download' api in Iuploader? Can't we use that if it exists?

Also to confirm this is a dataset resource not a static asset upload.

Reply:
The download API provides a response suitable for passing to Flask, typically a stream. However, what we need here is a URL to pass to Goodtables.

Reply:
Actually, a better option might be to use the standard resource download URL, `/resource/<UUID>/download`. I'm not sure why it was necessary to pass the direct S3 URL to Goodtables.

Reply:
does it need auth, how does it with for private resources?

Reply:
Private resources will likely succeed, since they'll get a signed S3 URL.